### PR TITLE
fix(streams): keep scale-drift exponentials in float32 domain

### LINKED ISSUE #421: ScaleDriftStream accepts reversed/non-finite log-scale bounds (dead drift, plausible stream); DynamicScaleShiftStream accepts non-positive/non-finite scale bounds (NaN observations)
## Summary

Two synthetic streams in `alberta_framework/streams/synthetic.py` accept scale bounds that make the stream measure something other than what its docstring says, without any error:

1. **`ScaleDriftStream`** — "Non-stationary stream where feature scales drift via random walk … `min_log_scale`: Minimum log-scale (clips random walk)". `__init__` performs no validation. With reversed bounds `jnp.clip(x, lo, hi)` degenerates to `hi`, so every scale is pinned and the advertised drift never happens — but the emitted stream is entirely finite and looks normal. The class exists specifically to measure "whether an online normalizer (EMANormalizer) can track scale shifts faster than Autostep's `v_i`"; with reversed bounds the arm silently measures a stationary-scale problem and the A/B is meaningless. Non-finite log bounds produce NaN observations.
2. **`DynamicScaleShiftStream`** — `min_scale <= 0`, non-finite bounds, or `min_scale > max_scale` are accepted; `jnp.log(min_scale)` then seeds NaN/inf scales and NaN/inf observations from an accepted config. `make_scale_range`'s log branch was already hardened for exactly this (#256) with `_require_normal_float32_scale`; the stream constructor was not.

## Falsifier (deterministic, pristine `main` @ `4f75cc8`)

```
=== ScaleDriftStream accepts min_log_scale > max_log_scale ===
  step 0: log_scales=[-4. -4. -4. -4.]  obs=[ 0.0062  0.0189 -0.0062  0.0102]
  step 3: log_scales=[-4. -4. -4. -4.]  obs=[ 0.0300 -0.0149  0.0155 -0.0200]
  -> every log scale pinned to max_log_scale=-4.0; the configured drift is dead
=== ScaleDriftStream accepts non-finite log-scale bounds ===
  obs: [nan nan nan]  target: [-2.878634]
=== DynamicScaleShiftStream accepts min_scale<=0 / non-finite ===
  {'min_scale': 0.0}:   init scales=[nan nan nan] obs=[nan nan nan]
  {'min_scale': -1.0}:  init scales=[nan nan nan] obs=[nan nan nan]
  {'min_scale': nan}:   init scales=[nan nan nan] obs=[nan nan nan]
  {'max_scale': inf}:   init scales=[inf inf inf] obs=[ inf  inf -inf]
```

Expected: `ScaleDriftStream` requires finite log bounds with `min_log_scale <= max_log_scale`; `DynamicScaleShiftStream` requires normal-float32 positive scale bounds (the existing helper) with `min_scale <= max_scale`.

## Dedup

Adjacent, closed: #256 (`make_scale_range(log_spaced=True)` bounds), #143 (zero schedule moduli), #249 (`FrequencyMismatchStream` omega), #269 (`CompositionalStream` weight_scale), #173 (`XDistShiftStream`), #298 (`ScaledStreamWrapper` ndim). `gh issue/pr list --search "ScaleDriftStream" / "DynamicScaleShiftStream" / "min_log_scale"`: nothing; no open PR touches `streams/synthetic.py`.

Fix PR follows (failing-test-first).

### PR DESCRIPTION
**Pull request by lalalune**
Supersedes #422 while preserving its reviewed scale-bound hardening on current `main`.

The original repair validated that log-scale bounds themselves remain finite in float32, but finite values such as `89.0` still overflow at the actual `jnp.exp` sink and values such as `-88.0` collapse to zero. This replacement additionally constrains the narrowed log bounds to the normal-float32 exponential domain before storing them, while retaining exact-ratio narrowing and the newer stream validation already on `main`.

Validation:
- `tests/test_streams.py`: 167 passed under `-W error`
- Ruff on changed source/tests: clean
- `git diff --check`: clean

No benchmark or evidence artifacts were modified.

Closes #421.

### CODE REVIEW

**Comment by lalalune** [comment]
Independent maintainer review: this is deeper and more correct than #618 because it validates the actual `jnp.exp(float32)` sink, not merely finite log bounds. I reproduced the exact accepted endpoints as positive/finite (`1.1754997e-38`, `3.4027977e38`) and the immediately adjacent float32 values as `0.0`/`inf`; the full stream suite passed under warnings-as-errors (167 tests) and Ruff is clean.

**Comment by lalalune** [comment]
Superseded by clean current-main PR #629. #629 carries the same exp-sink repair without the stale merge history and strengthens coverage to pin the exact accepted endpoints plus each immediately adjacent rejected float32 value.
